os/fs/smartfs/smartfs_utils.c

```diff
@@ -1333,47 +1316,47 @@ int smartfs_writeentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s new_
 	entry->utc = time(NULL);
 #endif
 #endif
+	smartfs_set_entry_flags(&new_entry, mode, type);
```

Comment:
if this is processed before writeentry, then we can prevent duplicated set.
Also mode & type can be removed from parameter.

Reply:
Updated